moveit_experimental/kinematics_cache/v2/kinematics_cache/src/kinematics_cache.cpp

```diff
@@ -152,23 +152,23 @@ bool KinematicsCache::addToCache(const geometry_msgs::Pose& pose, const std::vec
 
 bool KinematicsCache::getGridIndex(const geometry_msgs::Pose& pose, unsigned int& grid_index) const
 {
-  int x_index = (int)((pose.position.x - cache_origin_.x) / cache_resolution_x_);
-  int y_index = (int)((pose.position.y - cache_origin_.y) / cache_resolution_y_);
-  int z_index = (int)((pose.position.z - cache_origin_.z) / cache_resolution_z_);
+  int x_index = static_cast<int>((pose.position.x - cache_origin_.x) / cache_resolution_x_);
+  int y_index = static_cast<int>((pose.position.y - cache_origin_.y) / cache_resolution_y_);
+  int z_index = static_cast<int>((pose.position.z - cache_origin_.z) / cache_resolution_z_);
 
-  if (x_index >= (int)cache_size_x_ || x_index < 0)
+  if (x_index >= static_cast<int>(cache_size_x_ || x_index < 0))
   {
     ROS_DEBUG_NAMED("kinematics_cache", "X position %f,%d lies outside grid: %d %d", pose.position.x, x_index, 0,
                     cache_size_x_);
     return false;
   }
-  if (y_index >= (int)cache_size_y_ || y_index < 0)
+  if (y_index >= static_cast<int>(cache_size_y_ || y_index < 0))
   {
     ROS_DEBUG_NAMED("kinematics_cache", "Y position %f,%d lies outside grid: %d %d", pose.position.y, y_index, 0,
                     cache_size_y_);
     return false;
   }
-  if (z_index >= (int)cache_size_z_ || z_index < 0)
+  if (z_index >= static_cast<int>(cache_size_z_ || z_index < 0))
```

Comment:
Here are a few more where the parentheses change what is being cast.

Reply:
On to it. In a sec.